encode.go

```diff
@@ -92,34 +110,33 @@ func Encode(dst, src []byte) []byte {
 		if len(src) != 0 {
 			d += emitLiteral(dst[d:], src)
 		}
+		e.cur += 4
 		return dst[:d]
 	}
 
-	// Initialize the hash table. Its size ranges from 1<<8 to 1<<14 inclusive.
-	const maxTableSize = 1 << 14
-	shift, tableSize := uint(32-8), 1<<8
-	for tableSize < maxTableSize && tableSize < len(src) {
-		shift--
-		tableSize *= 2
+	// Ensure that e.cur doesn't wrap.
+	if e.cur > 1<<30 {
```

Comment:
`e.cur` relates to input position, but it is only used to invalidate entries from the previous time the encoder ran. When the inner loop is running, any reference to the previous block is discarded by the `t < 0` check.

`e.cur &= mask` is also a fine option, but it will always be a read+write, whereas the `if` can be `removed` by branch prediction - sine the branch will almost never be taken.